pkg/controller/node/controller_utils.go

```diff
@@ -36,17 +36,27 @@ import (
 )
 
 // This function is expected to get a slice of NodeReadyConditions for all Nodes in a given zone.
+// The zone is considered:
+// - fullyDisrupted if there're no Ready Nodes,
+// - partiallyDisrupted if more than 1/3 of Nodes (no less than 3) are not Ready,
+// - normal otherwise
 func ComputeZoneState(nodeReadyConditions []*api.NodeCondition) zoneState {
-	seenReady := false
+	readyNodes := 0
+	notReadyNodes := 0
 	for i := range nodeReadyConditions {
 		if nodeReadyConditions[i] != nil && nodeReadyConditions[i].Status == api.ConditionTrue {
-			seenReady = true
+			readyNodes++
+		} else {
+			notReadyNodes++
 		}
 	}
-	if seenReady {
-		return stateNormal
+	if readyNodes == 0 && notReadyNodes > 0 {
```

Comment:
nit: I'm not sure what do you think about it, but recently swtich commands became more readable for me in such cases. So personally I would change it to:
```
switch {
case readyNodes == 0 && notReadyNodes > 0:
  return stateFullDisruption
case notReadyNodes > 2 && 2*notReadyNodes > readyNodes
  return statePartialDisruption
default:
  return stateNormal
}
```
WDYT?